cartographer/pose_graph/proto/constraint.proto

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018 The Cartographer Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "cartographer/pose_graph/proto/loss_function.proto";
+import "cartographer/pose_graph/proto/cost_function.proto";
+
+package cartographer.pose_graph.proto;
+
+message Constraint {
+  string id = 1;
+  CostFunction cost_function = 2;
+  LossFunction loss_function = 3;
```

Comment:
Does this allow loss_function = empty? Or will this always default to Huber loss?

Reply:
Yes, it allows it to be empty. We can explicitly have an enum value for that inside  `LossFunction`.

Reply:
plz, take a look at `loss_function.proto`. I do not know yet how to express the default parabola explicity. Empty message?

Reply:
Yes, let's try the empty message 
message QuadraticLoss {}

If that works it would be nice and clean.